internal/install/recipe_installer.go

```diff
@@ -683,13 +691,99 @@ func (i *RecipeInstall) executeAndValidateWithProgress(ctx context.Context, m *t
 				i.progressIndicator.Canceled("Installing " + r.DisplayName)
 			} else {
 				i.progressIndicator.Fail("Installing " + r.DisplayName)
+				promptUserToSendErrorLogs(i, r.Name)
 			}
 			log.Debugf("install error encountered: %s", err)
 			return "", err
 		}
 	}
 }
 
+func promptUserToSendErrorLogs(i *RecipeInstall, recipeName string) {
+	if i.recipeExecutor.GetOutput().FailedRecipeOutput() != "" {
+		recipeOutputFilename := i.recipeExecutor.GetOutput().FailedRecipeOutput()
+
+		fmt.Printf("\n%s Something went wrong during the installation — let’s look at the docs and try again. Would you like to send us the logs to help us understand what happened? [Y/n] ", color.YellowString("\u0021"))
+		reader := bufio.NewReader(os.Stdin)
+		input, err := reader.ReadString('\n')
+		if err != nil {
+			log.Debugf("An error occurred while reading input: %e", err)
+		} else {
+			input = strings.TrimSuffix(input, "\n")
+			if strings.ToUpper(input) == "Y" {
+				postCliOutputToLogsAPI(i, recipeName)
+			}
+		}
+		os.Remove(recipeOutputFilename)
+	}
+}
+
+func postCliOutputToLogsAPI(i *RecipeInstall, recipeName string) {
+	// open file
+	f, err := os.OpenFile(i.recipeExecutor.GetOutput().FailedRecipeOutput(), os.O_RDONLY, os.ModePerm)
+	if err != nil {
+		log.Fatalf("open file error: %v", err)
+		return
+	}
+	defer f.Close()
+
+	// building log api client
+	config, err := nrLogClientConfig()
+	if nil != err {
+		log.Debugf("Could not configure New Relic LogsApi client: %e", err)
+		return
+	}
+	logClient := nrLogs.New(config)
+
+	sc := bufio.NewScanner(f)
+	fmt.Println()
+	i.progressIndicator.Start("Sending logs to New Relic")
+	// post each line to LogsApi
+	for sc.Scan() {
+		if err := logClient.CreateLogEntry(buildLogEntry(recipeName, sc.Text())); err != nil {
```

Comment:
Are we sending line by line? Might be fine, just wondering how quick this is.

Reply:
we are...though I don't think it would be too difficult to post all in one go